pandas/tests/window/moments/test_moments_ewm.py

```diff
@@ -53,63 +47,162 @@ def test_ewma_nan_handling():
     result = s.ewm(com=5).mean()
     tm.assert_series_equal(result, Series([np.nan] * 2 + [1.0] * 4))
 
-    # GH 7603
-    s0 = Series([np.nan, 1.0, 101.0])
-    s1 = Series([1.0, np.nan, 101.0])
-    s2 = Series([np.nan, 1.0, np.nan, np.nan, 101.0, np.nan])
-    s3 = Series([1.0, np.nan, 101.0, 50.0])
-    com = 2.0
-    alpha = 1.0 / (1.0 + com)
-
-    def simple_wma(s, w):
-        return (s.multiply(w).cumsum() / w.cumsum()).fillna(method="ffill")
-
-    for (s, adjust, ignore_na, w) in [
-        (s0, True, False, [np.nan, (1.0 - alpha), 1.0]),
-        (s0, True, True, [np.nan, (1.0 - alpha), 1.0]),
-        (s0, False, False, [np.nan, (1.0 - alpha), alpha]),
-        (s0, False, True, [np.nan, (1.0 - alpha), alpha]),
-        (s1, True, False, [(1.0 - alpha) ** 2, np.nan, 1.0]),
-        (s1, True, True, [(1.0 - alpha), np.nan, 1.0]),
-        (s1, False, False, [(1.0 - alpha) ** 2, np.nan, alpha]),
-        (s1, False, True, [(1.0 - alpha), np.nan, alpha]),
-        (s2, True, False, [np.nan, (1.0 - alpha) ** 3, np.nan, np.nan, 1.0, np.nan]),
-        (s2, True, True, [np.nan, (1.0 - alpha), np.nan, np.nan, 1.0, np.nan]),
```

Comment:
i'm not sure that the parameterisation is clearer.

maybe parameterise s, adjust and ignore_na independently (with fixtures) and create a `w` fixture with a dictionary lookup. i.e. pass s, adjust and ignore_na to `w` fixture

maybe use a class just for namespacing and use class level fixtures.

Reply:
I'll consider doing this change in a follow up PR. The main motivation with this pass is to ensure that parameterized test failures are more easily visible within pytest